x/mongo/driver/integration/opmsg_deployment.go

```diff
@@ -190,23 +190,23 @@ func (md *mockDeployment) Subscribe() (*driver.Subscription, error) {
 }
 
 // Unsubscribe is a no-op method which implements the driver.Subscriber interface.
-func (md *mockDeployment) Unsubscribe(*driver.Subscription) error {
+func (md *MockDeployment) Unsubscribe(*driver.Subscription) error {
 	return nil
 }
 
-// addResponses adds responses to this mock deployment.
-func (md *mockDeployment) addResponses(responses ...bson.D) {
+// AddResponses adds responses to this mock deployment.
+func (md *MockDeployment) AddResponses(responses ...bson.D) {
 	md.conn.responses = append(md.conn.responses, responses...)
 }
 
-// clearResponses clears all remaining responses in this mock deployment.
-func (md *mockDeployment) clearResponses() {
+// ClearResponses clears all remaining responses in this mock deployment.
+func (md *MockDeployment) ClearResponses() {
 	md.conn.responses = md.conn.responses[:0]
 }
 
-// newMockDeployment returns a mock driver.Deployment that responds with OP_MSG wire messages.
-func newMockDeployment(responses ...bson.D) *mockDeployment {
-	return &mockDeployment{
+// NewMockDeployment returns a mock driver.Deployment that responds with OP_MSG wire messages.
+func NewMockDeployment(responses ...bson.D) *MockDeployment {
```

Comment:
@joyjwang We should note in the comment that for most use cases it's better to test with an actual database. 

CC: @matthewdale